include/class.thread.php

```diff
@@ -993,14 +993,19 @@ function lookupByEmailHeaders(&$mailinfo, &$seen=false) {
                 //Lookup by ticket number
                 && ($ticket = Ticket::lookupByNumber($match[1]))
                 //Lookup the user using the email address
-                && ($user = User::lookup(array('emails__address' => $mailinfo['email'])))) {
+                && ($user = User::lookup(array('emails__address' => $mailinfo['email'])))
+                && ($staff = Staff::getIdByEmail($mailinfo['email']))) {
```

Comment:
It appears $user always tends to return on object even for staff members...  But you are right... there would be a bug if it was a user and not a staff member....
$staff = Staff::getIdByEmail($mailinfo['email']  should just be in the if statement, ill fix it